### PR TITLE
Add/Fix Dynamis CS IDs and Menu behavior

### PR DESCRIPTION
**Pull request by MarianArlt**
This is my original work from a few months ago rewritten from scratch. Since I now have retail access I could also double check on a few things first hand.

What this pull request fixes:  
- Most of all menus were using wrong IDs from obsolete Dynamis versions, hence those menus were not acting as expected
- Almost all Dynamis related areas had several dialogues, special messages and cutscenes missing
- Requirements for areas and general access was all over the place and not working as expected at all with lots of wrong checks

Some explanations:
- Variables were made more explicit to not be mistaken to be shorts for areas like Windy, Sandy etc.
- Menus are completely blocked behind general Dynamis requirements, which includes starter zone requirements
- Requirements for further zones are less complex as progress is cumulative and can be checked accordingly
- The 24h wait logic was outsourced into a local function to de-clutter entryNpcOnEventFinish
- A logic was added for the default trail markings dialogue and its exceptions
- First time CS now only plays after choosing to enter the zone, as it should, except Tavnazia, see code
- Exiting Dynamis by using the Somnial Treshold now plays the correct animation and then transports the player
- All text IDs that were added were adjusted for current state of base (that is: 3 to left from June 2021 Retail)

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits
(Last extensive testing that confirmed code was working as expected on July 9, upstream PR: 63b6e1b2)

### CODE REVIEW

File: scripts/globals/dynamis.lua [hunk 1]
```diff
@@ -292,37 +305,46 @@ dynamis.entryNpcOnTrigger = function(player, npc)
...
+
+        -- dynamis entry
+        elseif not info.reqs or info.reqs(player) then
+            local realDay = os.time()
+            local dynaWaitxDay = player:getCharVar("dynaWaitxDay")
+            local sjobOption = info.csBit > 6 and 1 or 0
+
+            if (dynaWaitxDay + xi.settings.BETWEEN_2DYNA_WAIT_TIME * 60 * 60) < realDay then
+                -- params: bit, cutscene option, Prismatic Hourglass KI, sJob option, junk, Shrouded Sand KI, Timeless Hourglass item ID, Perpetual Hourglass item ID
+                player:startEvent(info.csMenu, info.csBit, arg3(player, info.csBit), xi.ki.PRISMATIC_HOURGLASS, sjobOption, 0, xi.ki.VIAL_OF_SHROUDED_SAND, 4236, 4237)
```
Comment (MarianArlt) [inline scripts/globals/dynamis.lua:342]: The only real change is here is the addition of `sjobOption` to provide proper menu selections for cities and dreamlands alike. The rest of green comes from moving the default dialogue up.

**Comment by Era-Lusiphur** [comment]
Don't have a matching version so I can't test, but I like the new var names a lot.

**Review by claywar — APPROVED** [review]
Seems reasonable to me 👍  Text IDs might need adjustment once everything gets updated.

**Comment by claywar** [comment]
Merged this into its own branch so that Zach can make updates for text ID shifts.

**Comment by MarianArlt** [comment]
Oh sorry, I could've done that if ya liked. Was kinda observing whether or not that was requested. Feel free to do so in the future. FYI: I just checked retail Dynamis entries after yesterday's update (July) and it would _seem_ that, at least those, IDs did not shift this month.

**Comment by claywar** [comment]
About to merge in July text ID updates after CI passes on it; if there's no issue will yeet it in

**Comment by claywar** [comment]
Looks like there's a few conflicts to be resolved; if you want to hit those that'd be awesome, or one of us can

**Pull request merged**